pkg/controller/endpointslice/utils.go

```diff
@@ -267,6 +270,55 @@ func serviceControllerKey(endpointSlice *discovery.EndpointSlice) (string, error
 	return fmt.Sprintf("%s/%s", endpointSlice.Namespace, serviceName), nil
 }
 
+// setEndpointSliceLabels returns a map with the new endpoint slices labels and true if there was an update
+// The well-known labels IsHeadlessService, LabelServiceName and LabelManagedBy does not generate updates
```

Comment:
Maybe something like this
```suggestion
// Changes to IsHeadlessService, LabelServiceName and LabelManagedBy labels on the Service do not result in updates to EndpointSlice labels.
```